dawn/src/dawn/Optimizer/Replacing.cpp

```diff
@@ -79,8 +80,7 @@ void replaceFieldWithVarAccessInStmts(
 
       iir::replaceOldExprWithNewExprInStmt(stmt, oldExpr, newExpr);
 
-      metadata.insertExprToAccessID(newExpr, AccessID);
-      metadata.eraseExprToAccessID(oldExpr);
+      newExpr->getData<iir::IIRAccessExprData>().AccessID = std::make_optional(AccessID);
```

Comment:
I really don't like the interface. It might make sense internally, but acccessing doesn't make sense, because you wouldn't expect an optional (because you are never interested in whether accessID is set or not; if it is not set, it might be simply incorrect to access it, or it should be computed on the fly); and it doesn't make sense when setting the accessID, because it doesn't make sense to set nullopt

Reply:
So, this pattern is the same used for other data fields such as `CalleeAccesses`, `CallerAccesses` and `StackTrace` (even before the switch to data containers, where we used pointers instead of optional): if the datum is not yet calculated, the field is set to `nullptr`/`nullopt`. I agree that an interface around it would be nicer, but it would need to be public (can't `friend` every class that needs to write for example), and would necessarily consist of: `getAccessID()` (we already have `iir::getAccessIDFromExpr()`), `setAccessID(int)`, `unsetAccessID()` (needed by `AccessComputation.cpp:97` because a statement could be cloned and the clone brought out of a function, therefore `CalleeAccesses` must be unset), `isAccessIDSet()` ... which let's every user do the same things as exposing directly `AccessID`. Then I wonder if it is worth to switch to such interface and whether it is in the scope of this PR.

Reply:
Yes maybe not. If we add `iir::getAccessIDFromExpr`, we can hide most of them, so that's fine for me.